### DISCUSSION TITLE
Cannot open a ipynb file in a new environment

### DISCUSSION THREAD

**Discussion by eldar** [benlubas/molten-nvim#250] [Q&A]
Hi! Thanks for the awesome plugin. I've set up a virtual environment for Molten in the past, but due to the project requirements I needed to create a new one. In this new environment I can no longer open .ipynb files, all I get is an empty file. I've done the following from my new virtual environment, but neovim still doesn't show contents of the ipynb notebook file:

```bash
pip install pynvim jupyter_client cairosvg plotly kaleido pnglatex pyperclip
pip install ipykernel
python -m ipykernel install --user --name project_name
```
Given that it works on my old environment, there must be something missing in my configuration. The output of `:checkhealth molten` doesn't differ between the working and non-working environments:

```
molten-nvim ~
- OK NeoVim >=0.9
- OK Python >=3.10
- OK Python module pynvim found
- OK Python module jupyter-client found
- WARNING Optional python module cairosvg not found
  - ADVICE:
    - pip install cairosvg
- WARNING Optional python module pnglatex not found
  - ADVICE:
    - pip install pnglatex
- WARNING Optional python module plotly not found
  - ADVICE:
    - pip install plotly
- WARNING Optional python module kaleido not found
  - ADVICE:
    - pip install kaleido
- WARNING Optional python module pyperclip not found
  - ADVICE:
    - pip install pyperclip
- OK Python module nbformat found
- OK Python module pillow found
```

Could you please point me at how I can debug this? Thanks!

**Comment by benlubas** [comment]
Molten does not handle opening ipy files. This would be done by a plugin like jupytext.nvim and the command line utility jupytext (which I suspect is what's missing in the new env)

**Reply by eldar** [reply]
That was it, installing `jupytext` in my virtual env fixed it, thank you! Maybe worth adding this information to the documentation https://github.com/benlubas/molten-nvim/blob/main/docs/Virtual-Environments.md?

**Reply by benlubas** [reply]
I mean. Jupytext is not needed by this plugin in any way. Molten never uses it or checks for it.

There is information about it in the notebook setup docs, but it's made clear that it's part of a separate plugin.

That said, there are quite a good amount of people who assume that molten will handle the notebook conversion, so maybe it's worth explicitly stating that it doesn't.
